plugins/anghami/metadata.ini

```diff
@@ -0,0 +1,7 @@
+author=Mahmoud Hossam
+author_website=http://mahmoudhossam.com
+icon=anghami-logo.png
+name=Anghami
+require_proprietary_codecs=false
```

Comment:
Are you sure that service does not require proprietary codecs? On which platform did you tested it?

Reply:
I tested it on Vivaldi on Arch Linux, without the widevine plugin.

Is there a way to be sure it doesn't require any proprietary codecs?

Reply:
ffmpeg is the package that provides the proprietary codecs but you won't be able to remove it easily as it will also remove qtwebengine (and many other packages...)

I'd suggest to run your plugin from the MellowPlayer AppImage (the AppImage should find your plugin automatically).

Reply:
@ColinDuquesnoy I'll check out the AppImage, thanks.

Reply:
@ColinDuquesnoy I tried the AppImage and it didn't play apparently because pepper flash is required.

Is that considered a proprietary codec?

There's nothing much on this topic in the documentation, and other services don't seem to have this flag set at all.

Reply:
There are two cases to consider:

1. the service uses flash. In that case, yes, flash will provide the proprietary codecs. This is not an issue on most platforms because you just need to install pepper flash plugin (it will work from an AppImage). On ArchLinux, just install [pepper-flash](https://www.archlinux.org/packages/extra/x86_64/pepper-flash/)

2. The service use HTML 5 audio/video capabilities. In that case, proprietary codecs are provided by ffmpeg. It will not work in an AppImage (and on Windows and OSX) because QtWebEngine statically links with a version of ffmpeg compiled without proprietary codecs support (for licensing reasons).

There is a bit of documentation about that in the [readme](https://github.com/ColinDuquesnoy/MellowPlayer#limitations) and on the qtwebengine documentation. Apparently there is not enough documentation so I will include a section that talk about that in the developer documentation.

>  other services don't seem to have this flag set at all.

Mixcloud, soundcloud and spotify plugins set this flag to true (in the master branch). On the develop branch, I replace this flag by a list of supported platforms to avoid confusion (for end users). See #115.

Reply:
@ColinDuquesnoy I have pepper-flash installed on Arch Linux and the AppImage still does not play.

I get this warning in the AppImage output though:

```
[WARNING:flash/platform/pepper/pep_module.cpp(63)] SANDBOXED
```

Reply:
Ok so most likely the service needs codecs from ffmpeg (I assume the service works with the mellowplayer AUR package).

Reply:
Don't worry about the flag. I'll set it up myself.

Reply:
Don't worry about the flag. I'll set it up myself.